src/d3a/d3a_core/export.py

```diff
@@ -450,6 +503,43 @@ def __init__(self, dataset: dict, key: str):
         self.key = key
         self.dataset = dataset
         self.umHours = dict()
+        self.rate = list()
+        self.energy = list()
+
+    def sd_curve(self, supply=True):
+        try:
+            if supply:
+                sort_values = SortedDict(self.dataset)
+                self.rate = list(sort_values.keys())
+                self.energy = list(sort_values.values())
+            if not supply:
+                sort_values = SortedDict(self.dataset)
+                self.rate = list(reversed(sort_values.keys()))
+                self.energy = list(reversed(sort_values.values()))
+
+            cond_rate = list()
+            cond_energy = list()
+
+            for i in range(len(self.energy)):
+
+                if i == 0:
+                    cond_rate.append(self.rate[0])
+                    cond_energy.append(0)
+                    cond_rate.append(self.rate[0])
+                    cond_energy.append(self.energy[i])
+                else:
+                    if self.energy[i-1] == self.energy[i] and supply:
+                        continue
+                    cond_rate.append(self.rate[i])
```

Comment:
Sorry, can you please explain, what happens here? Again, there is duplication. Thanks!

Reply:
Same as above. Lets discuss offline.